src/createServer.js

```
@@ -0,0 +1,34 @@
+const express = require('express');
+const { authRouter } = require('./routes/auth.route.js');
+const cors = require('cors');
+const { userRouter } = require('./routes/user.route.js');
+const { errorMiddleware } = require('./middlewares/errorMiddleware.js');
+const cookieParser = require('cookie-parser');
+
+require('dotenv').config();
+
+const createServer = () => {
+  const app = express();
+
+  app.use(express.json());
+
+  app.use(cors({ origin: process.env.CLIENT_HOST, credentials: true }));
```

CORS is configured with `origin: process.env.CLIENT_HOST`. If `CLIENT_HOST` is undefined (development or misconfiguration), CORS may fail silently. Consider validating the env var or providing a development fallback (or explicit array of origins) and ensure `credentials: true` aligns with your cookie settings (secure/sameSite) on the client and server.